### PR TITLE
add mps support

### PR DESCRIPTION
**Pull request by maximegmd**
# Description

Supports basic training on Mac M series.

## Motivation and Context

It partially solves Mac support.

## How has this been tested?

Ran a train job with lora-mps.yml from start to finish.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -1,7 +1,9 @@
 """setup.py for axolotl"""
 
+import platform
 from importlib.metadata import PackageNotFoundError, version
 
+from packaging.version import Version, parse
```
Comment (winglian) [inline setup.py:6]: If we're going to require packaging, we probably have to update the docs so people install this before axolotl. So we'll have to update the ci to install packaging too.

**Comment by winglian** [comment]
@maximegmd I can help with this PR, but because your PR is from your `main` branch, it doesn't play nicely with github's maintainer tools. Could you submit this PR from a different branch from your fork please? This PR will be a great addition. Thanks!

File: setup.py (hunk 1 above)
Comment (maximegmd) [reply setup.py:6]: We already do say to install packaging first in the readme

**Comment by maximegmd** [comment]
Replaced by #1264

**Pull request closed without merging**